docs/examples/tutorial/string/c_func.py

```diff
@@ -0,0 +1,31 @@
+from cython.cimports.libc.stdlib import malloc
+from cython.cimports.libc.string import strcpy, strlen
+
+hello_world = cython.declare(cython.p_char, 'hello world')
+n = cython.declare(cython.Py_ssize_t, strlen(hello_world))
+
+
+# this function is defined as a cdef function in the .pxd file, no need for @cython.cfunc
+def c_call_returning_a_c_string() -> cython.p_char:
+    c_string: cython.p_char = cython.cast(cython.p_char, malloc(
+        (n + 1) * cython.sizeof(cython.char)))
+
+    if not c_string:
+        return cython.NULL  # malloc failed
+
+    strcpy(c_string, hello_world)
+    return c_string
+
+
+# this function is defined as a cdef function in the .pxd file, no need for @cython.cfunc
+@cython.cfunc  # DELETE ME (GH#5970)
+def get_a_c_string(c_string_ptr: cython.pp_char,
+                   length: cython.pointer(cython.Py_ssize_t)) -> cython.void:
```

Comment:
Why does this have to return `void`? Wouldn't it be more natural to return an integer status code (0/-1)?

Reply:
fixed.